database/postgresconn/postgresconn.go

```diff
@@ -281,26 +301,20 @@ func (p *Postgres) SetVersion(version int, dirty bool) error {
 		return &database.Error{OrigErr: err, Err: "transaction start failed"}
 	}
 
-	query := `TRUNCATE ` + pq.QuoteIdentifier(p.config.migrationsSchemaName) + `.` + pq.QuoteIdentifier(p.config.migrationsTableName)
-	if _, err := tx.Exec(query); err != nil {
-		if errRollback := tx.Rollback(); errRollback != nil {
-			err = multierror.Append(err, errRollback)
-		}
-		return &database.Error{OrigErr: err, Query: []byte(query)}
-	}
-
 	// Also re-write the schema version for nil dirty versions to prevent
 	// empty schema version for failed down migration on the first migration
 	// See: https://github.com/getoutreach/migrate/issues/330
-	if version >= 0 || (version == database.NilVersion && dirty) {
-		query = `INSERT INTO ` + pq.QuoteIdentifier(p.config.migrationsSchemaName) + `.` + pq.QuoteIdentifier(p.config.migrationsTableName) + ` (version, dirty) VALUES ($1, $2)`
-		if _, err := tx.Exec(query, version, dirty); err != nil {
-			if errRollback := tx.Rollback(); errRollback != nil {
-				err = multierror.Append(err, errRollback)
-			}
-			return &database.Error{OrigErr: err, Query: []byte(query)}
+	//if version >= 0 || (version == database.NilVersion && dirty) {
+	query := fmt.Sprintf(`INSERT INTO %q.%q`+
+		` (version, dirty, created_at) VALUES ($1, $2, now())`,
+		p.config.migrationsSchemaName, p.config.migrationsTableName)
```

Comment:
do we actually need two different forms of variable interpolation here (go Sprintf, and postgres vars)?

Might be worth a comment as to why, since I think it looks a little odd.

Reply:
Yes, this is normal, table and schema names cannot be done with bind variables, not the same as parameters, and need special quoting(shouldn't need document this)